services/debugger/cache.go

```diff
@@ -0,0 +1,100 @@
+package debugger
+
+import (
+	"github.com/rudderlabs/rudder-server/config"
+	"sync"
+	"time"
+)
+
+/*
+	loadCacheConfig sets the properties of the cache after reading it from the config file.
+This gives a feature of hot readability as well.
+ */
+func (cache *CacheT) loadCacheConfig() {
+	config.RegisterIntConfigVariable(3, &cache.size, true, 1, "LiveEvent.Cache.size")
+	config.RegisterDurationConfigVariable(time.Duration(720), &cache.keyTTL, true, time.Hour,
+		"LiveEvent.cache.ttl")            // default keyTTL is 30 days
+	config.RegisterDurationConfigVariable(time.Duration(15), &cache.cleanupFreq, true, time.Second,
+		"LiveEvent.cache.clearFreq") // default clearFreq is 15 seconds
+}
+
+type cacheItem struct {
+	data       [][]byte
+	lastAccess time.Time
+}
+
+/*
+	CacheT is an in-memory cache. Each key-value pair stored in this cache have a TTL and one goroutine removes the
+key-value pair form the cache which is older than TTL time.
+*/
+type CacheT struct {
```

Comment:
Not sure why the `T` prefix is added here, I am aware there is an old convention in the rudder-server but I haven't understood why we are doing it.

```suggestion
type Cache struct {
```

Reply:
This was changed just to adhere to the convention